runtime/src/lib.rs

```diff
@@ -321,31 +360,45 @@ type LocalAssetTransactor = xcm_builder::CurrencyAdapter<
 type LocalOriginConverter = (
 	SovereignSignedViaLocation<LocationConverter, Origin>,
 	RelayChainAsNative<RelayChainOrigin, Origin>,
-	SiblingParachainAsNative<cumulus_pallet_xcm_handler::Origin, Origin>,
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, Origin>,
 	SignedAccountId32AsNative<CentrifugeNetwork, Origin>,
 );
 
+pub type Barrier = (
+	TakeWeightCredit,
+	AllowTopLevelPaidExecutionFrom<All<MultiLocation>>,
+	AllowUnpaidExecutionFrom<All<MultiLocation>>, // TODO: Fees?
+);
+
+parameter_types! {
+	pub UnitWeightCost: Weight = 1_000;
+	pub const WeightPrice: (MultiLocation, u128) = (MultiLocation::X1(Junction::Parent), 1_000);
+}
 
 pub struct XcmConfig;
 impl Config for XcmConfig {
     type Call = Call;
-    type XcmSender = XcmHandler;
+    type XcmSender = XcmRouter;
     // How to withdraw and deposit an asset.
     type AssetTransactor = LocalAssetTransactor;
     type OriginConverter = LocalOriginConverter;
     type IsReserve = NativeAsset;
-    type IsTeleporter = ();
+    type IsTeleporter = (); // Don't allow any teleports
```

Comment:
Last I checked teleport will require hmp, which is currently worked on. Maybe we can enable it later on we do not have to pass through the relay chain

Reply:
Yeah, it should be easy enough to enable as part of actually getting xtokens functional